common/junit-platform-native/src/main/java/org/graalvm/junit/platform/config/platform/PlatformConfigProvider.java

```diff
@@ -0,0 +1,77 @@
+ * copyright rights in the Software, and any and all patent rights owned or
+ * freely licensable by each licensor hereunder covering either (i) the
+ * unmodified Software as contributed to or provided by such licensor, or (ii)
+ * the Larger Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ *
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.graalvm.junit.platform.config.platform;
+
+import org.graalvm.junit.platform.config.core.PluginConfigProvider;
+import org.graalvm.junit.platform.config.core.NativeImageConfiguration;
+
+public class PlatformConfigProvider implements PluginConfigProvider {
+
+    @Override
+    public void onLoad(NativeImageConfiguration registry) {
+        String[] buildTimeInitializedClasses = new String[]{
+                "org.junit.platform.launcher.core.InternalTestPlan",
+                "org.junit.platform.commons.util.StringUtils",
+                "org.junit.platform.launcher.core.TestExecutionListenerRegistry",
+                "org.junit.platform.commons.logging.LoggerFactory$DelegatingLogger",
+                "org.junit.platform.launcher.core.EngineDiscoveryOrchestrator",
+                "org.junit.platform.launcher.core.LauncherConfigurationParameters",
+                "org.junit.platform.commons.logging.LoggerFactory",
+                "org.junit.platform.engine.UniqueIdFormat",
+                "org.junit.platform.commons.util.ReflectionUtils"
+        };
+        for (String className : buildTimeInitializedClasses) {
+            registry.initializeAtBuildTime(className);
+        }
+
+        try {
+            registry.registerForReflection(Class.forName("org.junit.platform.commons.annotation.Testable").getMethods());
```

Comment:
`@Testable` does not declare any attributes (i.e. methods).

So why is `getMethods()` used here?

Reply:
That's a good question, I've moved this from earlier code but it does not seem necessary - we could only do a `Class.forName` here to see if the class is indeed present on the classpath.